test/test_setitem.py

```diff
@@ -4,28 +4,50 @@
 
 class TestSetitem(unittest.TestCase):
   def test_simple_setitem(self):
-    t = Tensor.zeros(6, 6).contiguous().realize()
-    t[2:4, 3:5] = Tensor.ones(2, 2)
-    n = np.zeros((6, 6))
-    n[2:4, 3:5] = np.ones((2, 2))
-    np.testing.assert_allclose(t.numpy(), n)
-
-    t = Tensor.zeros(6, 6).contiguous().realize()
-    t[2:4, 3:5] = 1.0
-    n = np.zeros((6, 6))
-    n[2:4, 3:5] = 1.0
-    np.testing.assert_allclose(t.numpy(), n)
+    cases = (
+      ((6,6), (slice(2,4), slice(3,5)), Tensor.ones(2,2)),
+      ((6,6), (slice(2,4), slice(3,5)), 1.0),
+      ((6,6), (slice(2,4), slice(3,5)), Tensor([1.,2.])),
+      ((4,4,4,4), (Ellipsis, slice(1,3), slice(None)), Tensor(4)),
+      ((4,4,4,4), (Ellipsis, slice(1,3)), 4),
+      ((4,4,4,4), (slice(1,3), slice(None), slice(0,4,2)), 4),
+      ((4,4,4,4), (slice(1,3), slice(None), slice(None), slice(0,3)), 4),
+      ((6,6), (slice(1,5,2), slice(0,5,3)), 1.0),
+      ((6,6), (slice(5,1,-2), slice(5,0,-3)), 1.0),
```

Comment:
added some cases here because `test_indexing` doesn't really test for non-advanced indexing setitem.